src/main/java/io/github/ryotan/code/util/CodeEnumReflectionUtil.java

```diff
@@ -0,0 +1,290 @@
+/*
+ * These codes are licensed under CC0.
+ *
+ * http://creativecommons.org/publicdomain/zero/1.0/deed
+ * http://creativecommons.org/publicdomain/zero/1.0/deed.ja
+ */
+
+package io.github.ryotan.code.util;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.AnnotatedElement;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+import io.github.ryotan.code.CodeEnum;
+import io.github.ryotan.code.CodeEnum.Filter;
+import io.github.ryotan.code.CodeEnum.ShortLabel;
+
+
+/**
+ * {@link CodeEnum}のリフレクション関連のユーティリティクラスです。
+ * <p>
+ * パラメータとして与えられたクラスが{@link CodeEnum}型のクラスである場合、コードパターンや短縮論理名を取得します。
+ * </p>
+ * <p>
+ * {@link Class}を引数に取るものは、与えられた{@link Class}がCodeEnum型でない場合には、{@link IllegalArgumentException}を送出します。
+ * </p>
+ *
+ */
+public final class CodeEnumReflectionUtil {
+    
+    /**
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Block comment has incorrect indentation level 4, expected is 3, indentation should be the same level as line 41.](https://www.codacy.com/app/ryotan/code-enum-support/file/3587099265/issues/source?bid=3375378&fileBranchId=3375797#l38)